views/js/previewer/provider/item.js

```diff
@@ -0,0 +1,388 @@
+ *
+ * Copyright (c) 2018 (original work) Open Assessment Technologies SA ;
+ */
+
+/**
+ * Test runner provider for the QTI item previewer
+ *
+ * @author Jean-Sébastien Conan <jean-sebastien@taotesting.com>
+ */
+define([
+    'jquery',
+    'lodash',
+    'i18n',
+    'core/store',
+    'core/promise',
+    'core/cachedStore',
+    'taoTests/runner/areaBroker',
+    'taoTests/runner/testStore',
+    'taoTests/runner/proxy',
+    'taoQtiTest/previewer/proxy/item',
+    'taoQtiTest/runner/ui/toolbox/toolbox',
+    'taoQtiItem/runner/qtiItemRunner',
+    'taoQtiTest/runner/config/assetManager',
+    'tpl!taoQtiTest/previewer/provider/item'
+], function (
+    $,
+    _,
+    __,
+    store,
+    Promise,
+    cachedStore,
+    areaBrokerFactory,
+    testStoreFactory,
+    proxyFactory,
+    proxyProvider,
+    toolboxFactory,
+    qtiItemRunner,
+    assetManagerFactory,
+    layoutTpl
+) {
+    'use strict';
+
+    //the asset strategies
+    var assetManager = assetManagerFactory();
+
+    proxyFactory.registerProvider('qtiItemPreviewerProxy', proxyProvider);
+
+    /**
+     * A Test runner provider to be registered against the runner
+     */
+    return {
+
+        //provider name
+        name: 'qtiItemPreviewer',
+
+        /**
+         * Initialize and load the area broker with a correct mapping
+         * @returns {areaBroker}
+         */
+        loadAreaBroker: function loadAreaBroker() {
+            var $layout = $(layoutTpl());
+
+            return areaBrokerFactory($layout, {
+                content: $('#qti-content', $layout),
+                toolbox: $('.bottom-action-bar .tools-box', $layout),
+                navigation: $('.bottom-action-bar .navi-box-list', $layout),
+                control: $('.top-action-bar .control-box', $layout),
+                actionsBar: $('.bottom-action-bar .control-box', $layout),
+                panel: $('.test-sidebar-left', $layout),
+                header: $('.top-action-bar .tools-box', $layout),
+                context: $('.top-action-bar .navi-box-list', $layout)
+            });
+        },
+
+        /**
+         * Initialize and load the test runner proxy
+         * @returns {proxy}
+         */
+        loadProxy: function loadProxy() {
+            var config = this.getConfig();
+            var proxyConfig = _.pick(config, [
+                'serviceCallId',
+                'bootstrap',
+                'timeout'
+            ]);
+
+            return proxyFactory(config.proxyProvider || 'qtiItemPreviewerProxy', proxyConfig);
+        },
+
+        /**
+         * Initialize and load the test store
+         * @returns {testStore}
+         */
+        loadTestStore : function loadTestStore(){
+            var config = this.getConfig();
+
+            //the test run needs to be identified uniquely
+            var identifier = config.serviceCallId || 'test-' + Date.now();
+            return testStoreFactory(identifier);
```

Comment:
do you need a test store for the item previewer ?

Reply:
Roughly I would say no, but some plugins do need it. Especially the theme switcher, that is mandatory if wish to keep the same features level as for the previous previewer.